hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DeadNodeHandler.java

```diff
@@ -74,12 +74,14 @@ public void onMessage(final DatanodeDetails datanodeDetails,
        * To be on a safer side, we double check here and take appropriate
        * action.
        */
-
       destroyPipelines(datanodeDetails);
       closeContainers(datanodeDetails, publisher);
 
-      // Remove the container replicas associated with the dead node.
-      removeContainerReplicas(datanodeDetails);
+      // Remove the container replicas associated with the dead node unless it
+      // is IN_MAINTENANCE
+      if (!nodeManager.getNodeStatus(datanodeDetails).isInMaintenance()) {
```

Comment:
Is it possible to get this message for a node `ENTERING_MAINTENANCE`?  If so, shouldn't it be ignored here as well?

Reply:
If a node goes dead while "entering maintenance" then it should be handled as a dead node. The reason, is that we have not yet determine if all the containers on the host are sufficiently replicated. If it goes dead before that check has completed and the node has moved into IN_MAINTENANCE, then it must be handled as a dead node as normal.

This is the same for a node which is DECOMMISSIONING and not yet reached DECOMMISSIONED.

Unless the node has reached its end state (IN_MAINTENANCE or DECOMMISSIONED), if it goes dead the workflow is aborted and it is handled just like any other dead node.